python/common/org/python/types/Str.java

```diff
@@ -972,12 +972,15 @@ private long toPositiveIndex(long index) {
         if (this.value.isEmpty()) {
             return new org.python.types.Bool(false);
         }
-        for (int c = 1; c < this.value.length(); c++) {
-            if (this.value.charAt(c - 1) == ' ' && !(Character.isUpperCase(this.value.charAt(c)))) {
-                return new org.python.types.Bool(false);
+
+        boolean matches = this.value.equals(_title(this.value));
+
```

Comment:
This works fine, but it could be better if you return false here if `matches` is false -- you only need to check if there is a letter if matches is `true`.